### PR TITLE
Replace b-buttons with buttons

### PR DESCRIPTION
**Pull request by fanglinfang**

### CODE REVIEW

File: myuw/static/vue/components/_templates/boilerplate/search.vue [hunk 1]
```diff
@@ -19,18 +19,15 @@
               placeholder="Search the UW website"
             />
             <b-input-group-append>
-              <b-button
-                variant="purple"
-                class="rounded-0"
-                type="submit"
+              <button class="btn btn-purple rounded-0" type="submit"
```
Comment (fanglinfang) [inline myuw/static/vue/components/_templates/boilerplate/search.vue:22]: do we have btn-purple defined?

File: myuw/static/vue/components/_templates/boilerplate/messages.vue [hunk 2]
```diff
@@ -19,7 +19,8 @@
         <b-link v-b-modal.tourModal class="text-white">
           <u>See MyUW at a glance</u>
         </b-link>
-        <b-button-close text-variant="light" @click="hideOnboardMessage" />
+        <button type="button" class="btn-close" aria-label="Close"
```
Comment (fanglinfang) [inline myuw/static/vue/components/_templates/boilerplate/messages.vue:22]: Char, this doesn't work. Any suggestion?

File: myuw/static/vue/components/_common/course/student/course.vue [hunk 3]
```diff
@@ -56,30 +56,26 @@
...
           v-if="section.is_ended || getSectionEval(section.index).length > 0"
         >
-          <b-button
-            v-b-toggle="`course-details-${index}`"
-            variant="link"
-            size="sm"
-            class="w-100 p-0 border-0 text-dark"
+          <button v-b-toggle="`course-details-${index}`"
+            type="button"
+            class="btn btn-link w-100 p-0 border-0 text-dark"
```
Comment (SarthakSingh31) [inline myuw/static/vue/components/_common/course/student/course.vue:61]: `btn-sm` is missing here

File: myuw/static/vue/components/_common/course/student/course.vue [hunk 4]
```diff
@@ -56,30 +56,26 @@
...
         <template v-else>
           <template v-if="section.instructors.length > 0">
-            <b-button
-              v-b-toggle="`instructors-collapse-${index}`"
-              variant="link"
-              size="sm"
-              class="w-100 p-0 border-0 text-dark"
+            <button v-b-toggle="`instructors-collapse-${index}`"
+              type="button"
+              class="btn btn-link w-100 p-0 border-0 text-dark"
```
Comment (SarthakSingh31) [inline myuw/static/vue/components/_common/course/student/course.vue:73]: `btn-sm` is missing here

File: myuw/static/vue/components/resources/resources.vue [hunk 5]
```diff
@@ -21,16 +21,14 @@
       </h2>
       <uw-resource-card :resource="resource" />
     </div>
-    <b-button
-      variant="secondary"
-      size="sm"
-      class="position-sticky mb-3 mr-3 float-right text-center myuw-back-to-top"
+    <button type="button"
+      class="btn btn-secondary position-sticky mb-3 mr-3 float-right text-center myuw-back-to-top"
```
Comment (SarthakSingh31) [inline myuw/static/vue/components/resources/resources.vue:25]: `btn-sm` is missing

File: myuw/static/vue/components/teaching/course/course_content.vue [hunk 6]
```diff
@@ -39,17 +39,15 @@
...
     </template>
     <template v-if="linkedSections.length > 0" #card-footer>
-      <b-button
-        v-b-toggle="`secondary-${section.section_label}`"
-        variant="link"
-        size="sm"
-        class="w-100 p-0 border-0 text-dark"
+      <button v-b-toggle="`secondary-${section.section_label}`"
+        type="button"
+        class="btn btn-link w-100 p-0 border-0 text-dark"
```
Comment (SarthakSingh31) [inline myuw/static/vue/components/teaching/course/course_content.vue:44]: `btn-sm` is missing

File: myuw/static/vue/components/_templates/boilerplate/search.vue (hunk 1 above)
Comment (SarthakSingh31) [reply myuw/static/vue/components/_templates/boilerplate/search.vue:22]: we do generate btn-purple.
Here are all our custom variants
https://github.com/uw-it-aca/myuw/blob/72071cfb03c4605ec3e0174de0874ec6a2ab53d3/myuw/static/css/myuw/variables.scss#L15-L48

File: myuw/static/vue/components/_templates/boilerplate/messages.vue (hunk 2 above)
Comment (SarthakSingh31) [reply myuw/static/vue/components/_templates/boilerplate/messages.vue:22]: This doesn't 1 to 1 translate. This provided by bootstrap-vue not base bootstrap.
It translates to
```html
<button type="button" aria-label="Close" class="close">×</button>
```
It also gets a event attached to it that I have not figured out yet. Best to leave this change out of this pr.

File: myuw/static/vue/components/_common/course/student/course.vue (hunk 4 above)
Comment (fanglinfang) [reply myuw/static/vue/components/_common/course/student/course.vue:73]: Looks good without btn-sm

File: myuw/static/vue/components/_common/course/student/course.vue (hunk 3 above)
Comment (fanglinfang) [reply myuw/static/vue/components/_common/course/student/course.vue:61]: "No instructor information available" has myuw-text-md

**Review by charlon — APPROVED** [review]

**Review by SarthakSingh31 — APPROVED** [review]

**Pull request merged**